sqs/src/main/scala/akka/stream/alpakka/sqs/SqsSourceStage.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.sqs
+
+import akka.stream.stage.{ GraphStage, GraphStageLogic, OutHandler }
+import akka.stream.{ Attributes, Outlet, SourceShape }
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.sqs.AmazonSQSAsyncClient
+import com.amazonaws.services.sqs.model.{ Message, ReceiveMessageRequest, ReceiveMessageResult }
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.concurrent.duration.FiniteDuration
+
+case class SqsSourceSettings(longPollingDuration: FiniteDuration, maxBufferSize: Int)
+
+class SqsSourceStage(queueUrl: String, settings: SqsSourceSettings, sqsClient: AmazonSQSAsyncClient) extends GraphStage[SourceShape[Message]] {
+
+  val out: Outlet[Message] = Outlet("SqsSource.out")
+  override val shape: SourceShape[Message] = SourceShape(out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = {
+    new GraphStageLogic(shape) {
+
+      private val buffer = mutable.Queue[Message]()
+
+      private val successCallback = getAsyncCallback[ReceiveMessageResult](handleSuccess)
+      private val failureCallback = getAsyncCallback[Exception](handleFailure)
+
+      def receiveMessages(): Unit = {
+
+        val request = new ReceiveMessageRequest(queueUrl)
+          .withMaxNumberOfMessages(10)
+          .withWaitTimeSeconds(settings.longPollingDuration.toSeconds.toInt)
+
+        sqsClient.receiveMessageAsync(request, new AsyncHandler[ReceiveMessageRequest, ReceiveMessageResult] {
```

Comment:
The threadpool get's configured on the cient itself not on the receive method. You are right that maybe users won't change the threadpool size. The only thin that could be changed in my opinion is to create an own AmazonSqsAsyncClient and configure it. (pushing the thread count as a setting into the Settings case class)